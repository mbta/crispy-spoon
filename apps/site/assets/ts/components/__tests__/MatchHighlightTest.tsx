import React from "react";
import { mount } from "enzyme";
import MatchHighlight from "../MatchHighlight";

describe("MatchHighlight", () => {
  it("renders without a match", () => {
    const wrapper = mount(
      <MatchHighlight
        text="As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system."
        matchQuery="zillion"
      />
    );
    wrapper.setProps({}); // actually updates the wrapper post-useEffect
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".u-highlight").exists()).toBeFalsy();
  });

  it("renders with a match", () => {
    const wrapper = mount(
      <MatchHighlight
        text="As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system."
        matchQuery="capital"
      />
    );
    wrapper.setProps({}); // actually updates the wrapper post-useEffect
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".u-highlight").exists()).toBeTruthy();
  });

  it("highlights only the first match if there are multiple matches", () => {
    const wrapper = mount(
      <MatchHighlight
        text="As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system."
        matchQuery="and"
      />
    );
    wrapper.setProps({}); // actually updates the wrapper post-useEffect
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".u-highlight")).toHaveLength(1);
  });

  it("highlights a query", () => {
    const wrapper = mount(
      <MatchHighlight
        text="As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system."
        matchQuery="billion"
      />
    );
    wrapper.setProps({}); // actually updates the wrapper post-useEffect
    expect(wrapper.find(".u-highlight").text()).toEqual("billion");
  });
});
