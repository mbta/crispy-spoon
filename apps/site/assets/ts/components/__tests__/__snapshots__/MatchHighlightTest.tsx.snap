// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchHighlight highlights only the first match if there are multiple matches 1`] = `"<span>As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, <b class=\\"u-highlight\\">and</b> ferries, and improving the accessibility of the entire system.</span>"`;

exports[`MatchHighlight renders with a match 1`] = `"<span>As part of our $8 billion, 5-year <b class=\\"u-highlight\\">capital</b> investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system.</span>"`;

exports[`MatchHighlight renders without a match 1`] = `"<span>As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system.</span>"`;
